imap_processing/swe/l1b/swe_l1b_science.py

```diff
@@ -0,0 +1,509 @@
+        L1A dataset
+
+    Returns
+    -------
+    xr.dataset
+        L1A dataset with filtered metadata.
+    """
+    for key, value in l1a_data.items():
+        l1a_data[key] = value.data[full_cycle_data_indices]
+    return l1a_data
+
+
+def swe_l1b_science(l1a_data):
+    """In this function, we cap.
+
+    Parameters
+    ----------
+    l1a_data : xr.dataset
+        L1A data
+    """
+    total_packets = len(l1a_data["SCIENCE_DATA"].data)
+
+    # Array to store list of table populated with data
+    # of full cycles
+    all_data = []
+    packet_index = 0
+    l1a_data_copy = l1a_data.copy(deep=True)
+
+    full_cycle_data_indices = get_indices_of_full_cycles(l1a_data["QUARTER_CYCLE"].data)
+
+    # Delete Raw Science Data from l1b and onwards
+    del l1a_data_copy["RAW_SCIENCE_DATA"]
+
+    if full_cycle_data_indices.size == 0:
+        # Log that no data is found for science data
+        return None
+
+    if len(full_cycle_data_indices) != total_packets:
+        # Filter metadata and science data of packets that makes full cycles
+        l1a_data_copy = filter_full_cycle_data(full_cycle_data_indices, l1a_data_copy)
+
+        # Update total packets
+        total_packets = len(full_cycle_data_indices)
+
+    # Go through each cycle and populate full cycle data
+    for packet_index in range(0, total_packets, 4):
+        # get ESA lookup table information
+        esa_table_num = l1a_data["ESA_TABLE_NUM"].data[packet_index]
+
+        # If ESA lookup table number is in-flight calibration
+        # data, then skip current cycle per SWE teams specification.
+        # SWE team only wants in-flight calibration data to be processed
+        # upto l1a. In-flight calibration data looks same as science data
+        # but it only measures one energy steps during the whole duration.
+        if esa_table_num == 1:
+            continue
+
+        full_cycle_data = populate_full_cycle_data(
+            l1a_data_copy, packet_index, esa_table_num
+        )
+
+        # save full data array to file
+        all_data.append(full_cycle_data)
+
+    # ------------------------------------------------------------------
+    # Save data to dataset.
+
+    # Get Epoch time of full cycle data and then reshape it to
+    # (n, 4) where n = total number of full cycles and 4 = four
+    # quarter cycle data metadata. For Epoch's data, we take the first element
+    # of each quarter cycle data metadata.
+    epoch_time = xr.DataArray(
+        l1a_data["Epoch"].data[full_cycle_data_indices].reshape(-1, 4)[:, 0],
+        name="Epoch",
+        dims=["Epoch"],
+        attrs=Epoch.output(),
+    )
+
+    # TODO: add more descriptive description
+    energy = xr.DataArray(
+        np.arange(24),
+        name="Energy",
+        dims=["Energy"],
+        attrs=dataclasses.replace(
+            swe_cdf_attrs.int_base,
+            catdesc="Energy",
+            fieldname="Energy",
+            label_axis="Energy",
+            units="keV",
```

Comment:
I changed it to this. Does that look ok?
```
energy = xr.DataArray(
        np.arange(180),
        name="Energy",
        dims=["Energy"],
        attrs=dataclasses.replace(
            swe_cdf_attrs.int_base,
            catdesc="Energy's index value in the lookup table",
            fieldname="Energy Bins",
            label_axis="Energy Bins",
            units="int",
        ).output(),
    )
```

Reply:
I don't think "int" is right, it should probably be unitless if that is allowed. Just an empty string maybe.

Reply:
Alternatively, could you actually pass in `decommutation_table` instead of `np.arange(180)` so you got the actual energy values here?

Reply:
I will leave empty string for now because those values in `decommutation_table` could be different based on look table id. I will keep this mind as we go forward.